airflow/utils/decorators.py

```diff
@@ -103,3 +103,19 @@ def wrapper(*args, **kwargs):
     # flake8: noqa: F811
     # Monkey patch hook to get good function headers while building docs
     apply_defaults = lambda x: x
+
+
+class cached_property:
+    """
+    A decorator who creates a property whose value is calculated only once.
+    """
+    def __init__(self, func):
+        self.func = func
+        self.__doc__ = getattr(func, '__doc__')
+
+    def __get__(self, instance, cls=None):
```

Comment:
The `cls` param seems unused. Is is necessary?

Reply:
It's a part of Python interface. Your suggestion works for me, but I do not know if it will work for any version of Python.